src/Jackalope/Transport/DoctrineDBAL/RepositorySchema.php

```diff
@@ -219,4 +224,30 @@ public function reset()
             $this->connection->exec($sql);
         }
     }
+
+    private function getMaxIndexLength($currentMaxLength = 255)
+    {
+        if (null !== $this->maxIndexLength) {
+            return $currentMaxLength < $this->maxIndexLength ? $currentMaxLength : $this->maxIndexLength;
+        }
+
+        $this->maxIndexLength = 255;
+
+        if ($this->isConnectionCharsetUtf8mb4()) {
+            $this->maxIndexLength = 191;
+        }
+
+        return $currentMaxLength < $this->maxIndexLength ? $currentMaxLength : $this->maxIndexLength;
+    }
+
+    private function isConnectionCharsetUtf8mb4()
+    {
+        if (!$this->connection) {
+            return false;
+        }
+
+        $databaseParameters = $this->connection->getParams();
```

Comment:
am i correct that this does not lead to a communication with the database? (otherwise we should be worried of the overhead)

Reply:
It only reads the [configuration](https://github.com/doctrine/dbal/blob/master/lib/Doctrine/DBAL/Connection.php#L245-L248) which is set in the [constructor](https://github.com/doctrine/dbal/blob/master/lib/Doctrine/DBAL/Connection.php#L206) of the dbal connection so this should not impact the performance.

Reply:
The only thing I don't know is why the connection is optional and not set here: https://github.com/jackalope/jackalope-doctrine-dbal/blob/1.3.1/src/Jackalope/Tools/Console/Command/InitDoctrineDbalCommand.php#L87

Reply:
i guess its optional so that the commands can be built at all when no connection is set up yet. and i guess in the command we did not pass the connection because it was not needed. happy if you change the command to pass the connection. is this also whats used in the phpcr bundle, or do we create the schema differently?

Reply:
did now also set the connection in InitDoctrineDbalCommand. In the phpcr bundle the connection seems correctly be injected in the [service definition](https://github.com/doctrine/DoctrinePHPCRBundle/blob/c9b8bbdbee712042f536b8a30902bb8614c2411c/src/Resources/config/jackalope_doctrine_dbal.xml#L14) and building the schema for sulu did work correctly for me.